### ISSUE TITLE
Usability improvements for playlist downloads

### ISSUE THREAD

**Issue by davidlav** [jely2002/youtube-dl-gui#98]
### Is your feature request related to a problem? Please describe.

**Download playlists into subdirectories**

Currently when downloading a playlist, all files are downloaded into the root of the specified download location.  So if, say, three playlists are downloaded into the same location, all the files are mixed together at the root level of that directory.  

**Sequentially enumerate files of playlists**

Additionally, no information is contained in the downloaded files that indicate the original ordering of the files in the playlist.

### Describe the solution you'd like

**Download playlists into subdirectories**

Either optionally, or by default, download playlists into individual subdirectories of the download location root directory, where the folders have the same name as the playlist.

So instead of 

```
Downloads/File from Playlist A
Downloads/Another file from Playlist A
Downloads/File from Playlist B
Downloads/Another file from Playlist B
```

you'd have

```
Downloads/Playlist A/File from Playlist A
Downloads/Playlist A/Another file from Playlist A
Downloads/Playlist B/File from Playlist B
Downloads/Playlist B/Another file from Playlist B
```

**Sequentially enumerate files of playlists**

For mp3s, add its position in the playlist to the Track field of its ID3 tag.

And also add the option for a sequential numeration prefixed to the filenames.  So instead of 

```
File A
File B
File C
```

it would be 

```
1. File A
2. File B
3. File C
```

or 

```
1 - File A
2 - File B
3 - File C
```

**Comment by jely2002** [comment]
Hi davidlav,

You can create your own file output formats, by using the `filename format` setting. 

![image](https://user-images.githubusercontent.com/20154900/121379934-c7ae9a80-c944-11eb-98b8-217b48f2a937.png)

So if you want a folder you could do `%(playlist_title)/$(title)s.%(ext)s`.
For documentation on this please check [youtube-dl](https://github.com/ytdl-org/youtube-dl#output-template).

### **Note:**
All playlist related placeholders currently do not work, see (#97). A fix has already been implemented and this will be released in the next bug fix or feature release.

**Comment by jely2002** [comment]
The playlist related placeholders have been fixed in v2.2.2. Closing this too as it is related.
